server/controllers/finance/accounts/accountReferenceType.js

```diff
@@ -0,0 +1,109 @@
+/**
+* Account Reference Type Controller
+*
+* This controller exposes an API to the client for reading and writing Account Reference Type
+*/
+
+const db = require('../../../lib/db');
+const NotFound = require('../../../lib/errors/NotFound');
+
+// GET /account_reference_type
+function lookupAccountReferenceType(id) {
+  const sql = `
+    SELECT id, label, fixed FROM account_reference_type
+    WHERE account_reference_type.id = ?`;
+
+  return db.one(sql, [id]);
+}
+
+
+// Lists
+function list(req, res, next) {
+  const sql = `SELECT id, label, fixed FROM account_reference_type;`;
+
+  db.exec(sql)
+    .then((rows) => {
+      res.status(200).json(rows);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* GET /account_reference_type/:ID
+*
+* Returns the detail of a single account_reference_type
+*/
+function detail(req, res, next) {
+  const { id } = req.params;
+
+  lookupAccountReferenceType(id)
+    .then((record) => {
+      res.status(200).json(record);
+    })
+    .catch(next)
+    .done();
+}
+
+
+// POST /account_reference_type
+function create(req, res, next) {
+  const sql = `INSERT INTO account_reference_type SET ?`;
+  const data = req.body;
+
+  db.exec(sql, [data])
+    .then((row) => {
+      res.status(201).json({ id : row.insertId });
+    })
+    .catch(next)
+    .done();
+}
+
+
+// PUT /account_reference_type /:id
+function update(req, res, next) {
+  const sql = `UPDATE account_reference_type SET ? WHERE id = ?;`;
```

Comment:
Let's delete `req.body.id` before doing this action to make sure the user cannot change the id.